src/utils/convert-filter.ts

```diff
@@ -8,36 +8,64 @@ const convertFilter = (filter) => {
     return {}
   }
   return filter.reduce((memo, filterProperty) => {
-    const { property, value } = filterProperty
+    const { property, value, path: filterPath } = filterProperty
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [_, index] = filterPath.split('.')
```

Comment:
Some more context:

When you pass an array to filter, it will generate:
`someField.0`: <filter data>
`someField.1`: <filter data>
`someField.2`: <filter data>

values in the filter and the previous version of the code was overwriting the filter instead of passing an array. I tried to use `flat.unflatten` on `filter` before using reduce so that it would be an actual array but it results in `maximum call stack size exceeded` error. We need to think of switching to another library/data model or optimising it a bit more...